contracts/governance/src/governance.rs

```diff
@@ -0,0 +1,274 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use core::iter;
+
+use alloc::vec;
+use alloc::vec::Vec;
+use canonical_derive::Canon;
+use dusk_bls12_381::BlsScalar;
+use dusk_hamt::Map;
+use dusk_jubjub::GENERATOR_EXTENDED;
+use dusk_pki::PublicKey;
+use dusk_schnorr::Signature;
+
+use crate::*;
+
+#[derive(Debug, Default, Clone, Canon)]
+pub struct GovernanceContract {
+    pub(crate) seeds: Map<BlsScalar, ()>,
+    pub(crate) balances: Map<PublicKey, u64>,
+    pub(crate) whitelist: Map<PublicKey, ()>,
+    pub(crate) paused: bool,
+    pub(crate) total_supply: u64,
+}
+
+impl GovernanceContract {
+    /// Authority of the contract
+    ///
+    /// Will have to be defined in the constant space so the bytecode of the
+    /// contract will be changed as the authority does
+    pub const AUTHORITY: PublicKey =
+        PublicKey::from_raw_unchecked(GENERATOR_EXTENDED);
```

Comment:
We could also use the reserved contract address space for this. If so, the authority would be part of the state of the contract - which makes it a little more flexible.
Just an idea, not a blocker for a merge.